pkg/distributor/distributor.go

```diff
@@ -1820,6 +1832,7 @@ func (d *Distributor) updateReceivedMetrics(req *mimirpb.WriteRequest, userID st
 	for _, ts := range req.Timeseries {
 		receivedSamples += len(ts.TimeSeries.Samples) + len(ts.TimeSeries.Histograms)
 		receivedExemplars += len(ts.TimeSeries.Exemplars)
+		d.costAttributionMgr.Tracker(userID).IncrementReceivedSamples(ts.Labels, float64(receivedSamples), mtime.Now())
```

Comment:
```suggestion
	cat := d.costAttributionMgr.Tracker(userID)
	for _, ts := range req.Timeseries {
		receivedSamples += len(ts.TimeSeries.Samples) + len(ts.TimeSeries.Histograms)
		receivedExemplars += len(ts.TimeSeries.Exemplars)
		cat.IncrementReceivedSamples(ts.Labels, float64(receivedSamples), mtime.Now())
```

Reply:
addressed in f8f2a49360e0028d56562b6da033fc0dd5c8f63a